src/pages/SupplierRequisitionPage.js

```diff
@@ -1,666 +1,334 @@
-                requisition.monthsToSupply = number;
-                database.save('Requisition', requisition);
-              });
-              this.refreshData();
-              this.closeModal();
-            }}
-            selected={requisition.monthsToSupply}
-          />
-        );
-      case COMMENT_EDIT:
-        return (
-          <TextEditor
-            text={requisition.comment}
-            onEndEditing={newComment => {
-              if (newComment !== requisition.comment) {
-                database.write(() => {
-                  requisition.comment = newComment;
-                  database.save('Requisition', requisition);
-                });
-              }
-              this.closeModal();
-            }}
-          />
-        );
-      case VIEW_REGIMEN_DATA:
-        return (
-          <RequisitionRegimenModalTable
-            database={database}
-            requisition={requisition}
-            genericTablePageStyles={genericTablePageStyles}
-          />
-        );
+        return null;
     }
-  };
-
-  renderButtons = () => {
-    const { requisition } = this.props;
-    const { program, thresholdMOS, parsedCustomData } = requisition;
-    const hasRegimenData =
-      parsedCustomData && parsedCustomData.regimenData && parsedCustomData.regimenData.length !== 0;
-    const UseSuggestedQuantitiesButton = () => (
-      <PageButton
-        style={{
-          ...globalStyles.topButton,
-          ...(program ? { marginLeft: 5 } : {}),
-        }}
-        text={buttonStrings.use_suggested_quantities}
-        onPress={this.onUseSuggestedQuantities}
-        isDisabled={requisition.isFinalised}
-      />
-    );
-
-    const ViewRegimenDataButton = () => (
-      <PageButton
-        style={{
-          ...globalStyles.topButton,
-          ...(program ? { marginLeft: 5 } : {}),
-        }}
-        text={buttonStrings.view_regimen_data}
-        onPress={() => this.openModal(MODAL_KEYS.VIEW_REGIMEN_DATA)}
-      />
-    );
-
-    const ThresholdMOSToggle = () => {
-      const { useThresholdMOS } = this.state;
-      const onPress = () => this.setState({ useThresholdMOS: !useThresholdMOS }, this.refreshData);
-      const toggleProps = [
-        {
-          text: programStrings.hide_over_stocked,
-          isOn: useThresholdMOS,
-          onPress,
-        },
-        {
-          text: programStrings.show_over_stocked,
-          isOn: !useThresholdMOS,
-          onPress,
-        },
-      ];
-      return <ToggleBar style={globalStyles.toggleBar} toggles={toggleProps} />;
-    };
+  });
+
+  const renderRow = useCallback(
+    listItem => {
+      const { item, index } = listItem;
+      const rowKey = keyExtractor(item);
+      const { row, alternateRow } = newDataTableStyles;
+      return (
+        <DataTableRow
+          rowData={data[index]}
+          rowState={dataState.get(rowKey)}
+          rowKey={rowKey}
+          style={index % 2 === 0 ? alternateRow : row}
```

Comment:
this snippet is already used in a few places and looks like a good candidate for pushing down into the `DataTableRow` component. perhaps pass the index as a prop and have another prop which colours alternate rows, defaulting to true so that you don't have to pass it in?